Modules/Devices/src/Devices.Application/Identities/Queries/GetDeletionProcess/Handler.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Backbone.BuildingBlocks.Application.Abstractions.Exceptions;
+using Backbone.BuildingBlocks.Application.Abstractions.Infrastructure.UserContext;
+using Backbone.Modules.Devices.Application.DTOs;
+using Backbone.Modules.Devices.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Devices.Domain.Entities.Identities;
+using MediatR;
+
+namespace Backbone.Modules.Devices.Application.Identities.Queries.GetDeletionProcess;
+
+public class Handler : IRequestHandler<GetDeletionProcessQuery, IdentityDeletionProcessDTO>
+{
+    private readonly IIdentitiesRepository _identitiesRepository;
+    private readonly IUserContext _userContext;
+
+    public Handler(IIdentitiesRepository identitiesRepository, IUserContext userContext)
+    {
+        _identitiesRepository = identitiesRepository;
+        _userContext = userContext;
+    }
+
+    public async Task<IdentityDeletionProcessDTO> Handle(GetDeletionProcessQuery request, CancellationToken cancellationToken)
+    {
+        var identity = await _identitiesRepository.FindByAddress(_userContext.GetAddress(), cancellationToken) ?? throw new NotFoundException(nameof(Identity));
+        var deletionProcess = identity.DeletionProcesses.FirstOrDefault(p => p.Id == request.Id) ?? throw new NotFoundException(nameof(IdentityDeletionProcess));
```

Comment:
This approach certainly works, but maybe it's a bit of an overkill to fetch the entire identity and all its deletion processes just to get the one you need... On the other hand, I can imagine that there will rarely be an identity with more than one process, and probably never with a significant number of them.

I will not request changes here so that @tnotheis may have the final word on this.

Reply:
I agree that it is not ideal. But on the other hand it's as you said: there won't be that much data, so I don't think it is worth the time we spend on implementing something better. 
Also, we would have to pollute the Identity class as well as the IdentityRepository to achieve this.
So in short: we will leave it as it is.